src/compile/unit.ts

```diff
@@ -271,7 +272,15 @@ export class UnitModel extends ModelWithField {
       marks = assembleUnitSelectionMarks(this, marks);
     }
 
-    return marks.map(this.correctDataNames);
+    const marksWithCorrectedDataNames = marks.map(this.correctDataNames);
+    if ('time' in this.encoding) {
+      marksWithCorrectedDataNames.forEach(mark => {
+        if (mark.from?.data) {
+          mark.from.data = mark.from.data + CURR;
```

Comment:
Why does this happen here? Could/should it happen in `this.correctDataNames` (which, besides keeping mark-data naming all encapsulating in one function, might also benefit other parts of the codebase?).

Reply:
I moved `correctDataNames` into unit model since it requires inspecting `this.encoding` and it's only used for this method.

Reply:
resolved in https://github.com/vega/vega-lite/pull/8921/commits/3b63808eaf1d31e09d6cd203164e39ac2a28e711